### PR TITLE
Update github actions ci

### PR DESCRIPTION
**Pull request by 4144**
- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

Add gcc-snapshot into github actions ci.

### CODE REVIEW

File: src/common/memmgr.h [hunk 1]
```diff
@@ -70,18 +70,21 @@
 #define CREATE(result, type, number) ((result) = (type *) aCalloc((number), sizeof(type)))
 #define RECREATE(result, type, number) ((result) = (type *) aReallocz((result), sizeof(type) * (number)))
 
+#ifdef HERCULES_CORE
```
Comment (MishimaHaruna) [inline src/common/memmgr.h:73]: This looks like an empty ifdef block

File: src/common/memmgr.h [hunk 2]
```diff
@@ -92,14 +95,29 @@ struct malloc_interface {
...
+
 void malloc_defaults(void);
 
 void memmgr_report(int extra);
 
 HPShared struct malloc_interface *iMalloc;
 #else
+
+#ifndef iMalloc
+#define iMalloc HPMi->memmgr
```
Comment (MishimaHaruna) [inline src/common/memmgr.h:114]: The same define looks like it's duplicated below

File: src/map/clif.c [hunk 3]
```diff
@@ -9951,8 +9951,13 @@ static void clif_mobname_additional_ack(int fd, struct block_list *bl)
 		str_p += sprintf(str_p, "HP: %u%% | ", get_percentage(md->status.hp, md->status.max_hp));
 	//Even thought mobhp ain't a name, we send it as one so the client
 	//can parse it. [Skotlex]
-	if (str_p != mobhp) {
-		*(str_p-3) = '\0'; //Remove trailing space + pipe.
+	if ((battle_config.show_mob_info & (4 + 1 + 2)) != 0 && str_p != mobhp) {
```
Comment (MishimaHaruna) [inline src/map/clif.c:9954]: The result is the same, but I think this is more error-proof if/when adding new values in the future:

```suggestion
	if ((battle_config.show_mob_info & (4 | 1 | 2)) != 0 && str_p != mobhp) {
```

**Review by MishimaHaruna — CHANGES_REQUESTED** [review]

**Comment by 4144** [comment]
apply review comments and rebased

**Review by MishimaHaruna — APPROVED** [review]

**Pull request merged**